From a1884e05979783bf66354078edf6e4eb1ad9b9d4 Mon Sep 17 00:00:00 2001
From: Chung-Hsien Hsu <stanley.hsu@cypress.com>
Date: Thu, 20 Jun 2019 02:59:26 -0500
Subject: [PATCH 050/101] brcmfmac: fix credit reserve for each access category

Commit 7f52880c5ecd ("brcmfmac: Allow credit borrowing for all access
categories") added a ratio to reserve the credits for each access
category. However, calculating the number of reserved credits with the
runtime one let an access category lend all its credits. Fix this by
using the number of initial credits to determine the reserved one.

Signed-off-by: Chung-Hsien Hsu <stanley.hsu@cypress.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/fwsignal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fwsignal.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fwsignal.c
index 74a6272..030452c 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fwsignal.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fwsignal.c
@@ -2036,7 +2036,7 @@ static int brcmf_fws_borrow_credit(struct brcmf_fws_info *fws,
 
 		if (!borrow_all)
 			borrow_limit =
-			  fws->fifo_credit[lender_ac] / BRCMF_BORROW_RATIO;
+			  fws->init_fifo_credit[lender_ac] / BRCMF_BORROW_RATIO;
 		else
 			borrow_limit = 0;
 
-- 
1.9.1

